src/pages/products-page/ProductCard.jsx

```diff
@@ -0,0 +1,52 @@
+import { useProduct } from "../../product-context";
+
+export default function ProductCard({ item }) {
+  const { state, dispatch } = useProduct();
+
+  return (
+    <div class="h-grid-product" key={item._id}>
+      <img src={item.image} />
+      <div class="h-product-icon">
+        {state.wishList.some((p) => p._id === item._id) ? (
+          <i
+            class="fas fa-heart"
+            onClick={() =>
+              dispatch({ type: "REMOVE_FROM_WISHLIST", payload: item })
```

Comment:
Nicely structured code, but whenever you are dispatching an action you have to remember exact wordings of `type` rather you could store it in some object and use that object wherever needed, eg: `ACTIONS.AddToCart`